controller/filehandlingcontroller.php

```diff
@@ -198,6 +200,9 @@ public function save($path, $filecontents, $mtime) {
 				// Get file mtime
 				$filemtime = $node->getMTime();
 
+				// Get lock (if there is any)
+				$activePersistentLock = $this->getPersistentLock($node);
```

Comment:
Handling this in save is too late - the user already made changes which can not not be written and are potentially lost.

The correct workflow is:
- try to lock the file when opening it
- if the lock cannot be acquired -> user is not allowed to edit the file and it shall be presented in read only mode
- if lock was acquired -> user is allowed to edit
- save is from now on any time possible
- closing the file unlocks the file

Reply:
ok, need to explore this, was not aware texteditor allows read-only mode.

Reply:
It doesn't. Has to be implemented 🤪

Reply:
@DeepDiver1975 just tested, if you share a file with read-only permission, it goes in a weird freeze mode that has cursor but does not allow input 😆 

However, what you describe is implementing locking in text editor. In my PR, it only respects lock and does not allow edit if it finds one. To implement locking in text editor it is a bit more complicated and more work -> was not a requirement. But what I definitely can explore is that if lock is found on open then it should be read-only.

Reply:
Maybe when opening a file in text editor, the code can check if it has no write access (is read-only or file is already locked).
If there is no write access, then just do something to render the file content (ordinary lines of text) and provide a close button somewhere, and some notification that edit is not available - don't even open the "text editor element".

Reply:
@phil-davis yes - this is exactly the behavior I am proposing above. :+1:

Reply:
Yes I am akready on it, finalizing tests